### DISCUSSION TITLE
Skipped jobs should provide reason

### DISCUSSION THREAD

**Discussion by nathaniel-palmer-simplisafe** [community/community#60882] [Actions]
### Select Topic Area

Product Feedback

### Body

Currently when a job is skipped due to an [if conditional](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif) the output view just says "This job was skipped" with no further explanation. This can be confusing for users who may be opening PRs but are not familiar with GitHub actions and may be confused why an expected check is not being run.

**Comment by brunokc** [comment]
I agree. I'm trying to debug a situation where a job with an if conditional is not running. GitHub output provides no help, even when debugging is enabled. I've added artificial steps and jobs to try to debug the problem and it all works as expected except for the job with the if conditional, which is not running for some reason. Since there's on debugging output for skipped jobs, I'm blocking from making progress. I'll likely have to stop using the if conditional in favor if using `if` inside `run`, if applicable.

**Comment by holiveros-samsung** [comment]
Similar situation here, got a deployment step which is being skipped with no clear explanation why. Step calls a reusable workflow and has a dependency on a dummy step just to pass env vars (another limitation of GH Actions).

**Comment by markfickett** [comment]
Similar discussion with more votes/participation here: https://github.com/orgs/community/discussions/20640
